core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-toolbar/dot-uve-toolbar.component.ts

```diff
@@ -81,15 +88,43 @@ export class DotUveToolbarComponent {
         return 'uve-toolbar uve-toolbar-preview';
     });
 
-    protected readonly date = new Date();
+    protected readonly publishDateParam = this.#store.pageParams().publishDate;
+    protected readonly $previewDate = model<Date>(
+        this.publishDateParam ? new Date(this.publishDateParam) : null
+    );
+
+    readonly $previewDateEffect = effect(
+        () => {
+            const previewDate = this.$previewDate();
+
+            if (!previewDate) {
+                return;
+            }
+
+            // If previewDate is minor that the CURRENT DATE, set previewDate to CURRENT DATE
+            if (previewDate < this.CURRENT_DATE) {
+                this.$previewDate.set(this.CURRENT_DATE);
+
+                return;
+            }
+
+            untracked(() => {
+                this.#store.loadPageAsset({
+                    preview: 'true',
+                    publishDate: previewDate?.toISOString()
+                });
+            });
+        },
+        { allowSignalWrites: true }
+    );
```

Comment:
Why this is needed?

Reply:
Im actually having the same issue with the view params

Reply:
Update: It was kinda the same problem, but I solved it for viewParams